pkg/store/bucket.go

```diff
@@ -945,12 +950,25 @@ func populateChunk(out *storepb.AggrChunk, in chunkenc.Chunk, aggrs []storepb.Ag
 			if err != nil {
 				return err
 			}
-			out.Counter = &storepb.Chunk{Type: storepb.Chunk_XOR, Data: b}
+			out.Counter = &storepb.Chunk{Type: storepb.Chunk_XOR, Data: b, Hash: hashChunk(b, calculateChecksum)}
 		}
 	}
 	return nil
 }
 
+func hashChunk(b []byte, doHash bool) uint64 {
+	hasher := hashPool.Get().(hash.Hash64)
+	defer hashPool.Put(hasher)
```

Comment:
Probably a better way to go is to extract these 2 lines at the beginning of `populateChunk` function. This can reuse   `hash.Hash64` and avoid some lock contention of the pool.